pkg/actuators/machine/machine.go

```diff
@@ -54,80 +55,250 @@ func newMachineScope(
+		}
+	}
+
+	instance, err := ms.ovirtClient.CreateVM(ovirtC.ClusterID(clusterId),
+		temp.ID(),
 		ms.machine.Name,
-		ms.machine.Labels["machine.openshift.io/cluster-api-cluster"],
-		ignition,
-		ms.machineProviderSpec)
+		optionalVMParams, ovirtC.ContextStrategy(ms.Context))
+
 	if err != nil {
 		return errors.Wrap(err, "error creating Ovirt instance")
 	}
 
 	// Wait till ready
-	err = ms.waitTillVMReachStatus(ovirtsdk.VMSTATUS_DOWN)
+	_, err = ms.ovirtClient.WaitForVMStatus(instance.ID(), ovirtC.VMStatusDown, ovirtC.ContextStrategy(ms.Context))
 	if err != nil {
 		return errors.Wrap(err, "error creating oVirt VM")
 	}
 
+	var bootableDiskAttachment ovirtC.DiskAttachment
+	newDiskSize := uint64(ms.machineProviderSpec.OSDisk.SizeGB * int64(math.Pow(2, 30)))
+	// Handle OS disk extension
+	if ms.machineProviderSpec.OSDisk != nil {
+		diskAttachments, err := instance.ListDiskAttachments()
+		if err != nil {
+			return errors.Wrapf(err, "Failed to list disk attachments for VM %s.", instance.ID())
+		}
+		for _, diskAttachment := range diskAttachments {
+			if diskAttachment.Bootable() {
+				bootableDiskAttachment = diskAttachment
+			}
+		}
+
+		if bootableDiskAttachment == nil {
+			return errors.Wrapf(err, "VM %s(%s) doesn't have a bootable disk", instance.Name(), instance.ID())
+		}
+
+		disk, err := ms.ovirtClient.GetDisk(bootableDiskAttachment.DiskID(), ovirtC.ContextStrategy(ms.Context))
+		if newDiskSize > disk.ProvisionedSize() {
+			updatedDisk, err := disk.Update(ovirtC.UpdateDiskParams().MustWithProvisionedSize(newDiskSize))
+			if err != nil {
+				return errors.Wrapf(err, "Failed to extend disk %s (%v)", disk.ID())
+			}
+			klog.Infof("Waiting for disk to become OK...")
+			updatedDisk, err = updatedDisk.WaitForOK()
+			if err != nil {
+				return errors.Wrapf(err, "Failed to wait for disk %s to return to OK status. (%v)", disk.ID())
+			}
+		}
+	}
+
+	// handleNics reattachment
+	if ms.machineProviderSpec.NetworkInterfaces != nil && len(ms.machineProviderSpec.NetworkInterfaces) > 0 {
+		nics, err := instance.ListNICs()
+		if err != nil {
+			return errors.Wrapf(err, "failed to list NICs on VM %s", instance.ID())
+		}
+
+		//remove all the nics from the VM instance
+		for _, nic := range nics {
+			if err := nic.Remove(); err != nil {
+				return errors.Wrapf(err, "failed to remove NIC %s", nic.ID())
+			}
+		}
+
+		//re-create NICs According to the machinespec
+		for i, nic := range ms.machineProviderSpec.NetworkInterfaces {
+			_, err := instance.CreateNIC(fmt.Sprintf("nic%d", i+1), nic.VNICProfileID, ovirtC.CreateNICParams())
+
+			if err != nil {
+				return errors.Wrap(err, "failed to create network interface")
+			}
+		}
+	}
+
+	if isAutoPinning {
+		err = ms.ovirtClient.AutoOptimizeVMCPUPinningSettings(instance.ID(), true, ovirtC.ContextStrategy(ms.Context))
+		if err != nil {
+			return errors.Wrapf(err, "failed to Optimize CPU pinning settings for VM %s", instance.ID())
```

Comment:
lower case "optimize"? Also, I believe we can just return `err`. go-ovirt-client should take care of producing reasonable errors.